.github/workflows/upload_component.yml

```diff
@@ -0,0 +1,20 @@
+name: Push components to Espressif Component Service
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  upload_components:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```

Comment:
Is there a reason for using this old version of checkout action? I would suggest using v2.  `actions/checkout@v2`  which is more flexible.

Reply:
`checkout@v2` requires git version >= 2.18 if submodules are fetched (https://github.com/actions/checkout#checkout-v2).

in idf docker image v4.3 this is not met :(

it is fixed in v4.4 https://github.com/espressif/esp-idf/commit/57b243a6999c8b1a4c31cf8ce6e6d715c51bb8c3

Reply:
Ok I understand this limitation  for `build_test.yml` workflow
But in this workflow, it runs directly in VM `runs-on: ubuntu-latest`, so it shouldn't be a problem.
It's ok however, to keep v1 here too.

Reply:
True, fixed